src/index.test.ts

```diff
@@ -0,0 +1,448 @@
+    storage.clear();
+  });
+
+  it("should set and get an item", () => {
+    storage.setItem("plop", "coucou");
+    expect(storage.getItem("plop")).toEqual("coucou");
+  });
+
+  it("should clear the store", () => {
+    storage.setItem("plop", "coucou");
+    storage.clear();
+
+    expect(store).toEqual({});
+  });
+
+  it("should return null if the value is not in the store", () => {
+    expect(storage.getItem("plop")).toBeNull();
+  });
+});
+
+describe("react-runtime-config", () => {
+  const namespace = "test";
+  const createConfigWithDefaults = (
+    config: Pick<Partial<ConfigOptions<never, "">>, "localOverride" | "namespace"> = {},
+  ) =>
+    createConfig({
+      namespace,
+      storage,
+      schema: {
+        color: {
+          type: "string",
+          enum: ["blue" as const, "green" as const, "pink" as const],
+          description: "Main color of the application",
+        },
+        backend: {
+          type: "string",
+          description: "Backend url",
+        },
+        port: {
+          type: "number",
+          description: "Backend port",
+          min: 1,
+          max: 65535,
+          default: 8000,
+        },
+        monitoringLink: {
+          type: "custom",
+          description: "Link of the monitoring",
+          parser: value => {
+            if (typeof value === "object" && typeof value.url === "string" && typeof value.displayName === "string") {
+              return {
+                url: value.url as string,
+                displayName: value.displayName as string,
+              };
+            }
+            throw new Error("Monitoring link invalid!");
+          },
+        },
+        isLive: {
+          type: "boolean",
+          default: false,
+        },
+        isAwesome: {
+          type: "boolean",
+        },
+      },
+      ...config,
+    });
+
+  beforeEach(() => {
+    set(window, namespace, {
+      color: "blue",
+      backend: "http://localhost",
+      monitoringLink: {
+        url: "http://localhost:5000",
+        displayName: "Monitoring",
+      },
+      isAwesome: true,
+    });
+  });
+  afterEach(() => {
+    act(() => storage.clear());
+    delete (window as any)[namespace];
+  });
+
+  it("should throw if a window value don't fit the schema", () => {
+    set(window, `${namespace}.color`, "red");
+    expect(() => createConfigWithDefaults()).toThrowErrorMatchingInlineSnapshot(
+      `"Config key \\"color\\" not valid: red not part of [\\"blue\\", \\"green\\", \\"pink\\"]"`,
+    );
+  });
+
+  describe("getConfig", () => {
+    it("should return the default value", () => {
+      const { getConfig } = createConfigWithDefaults();
+      expect(getConfig("port")).toBe(8000);
+    });
+
+    it("should return the default value (function)", () => {
+      let port = 8000;
+      const getPort = () => port;
+      const { getConfig } = createConfig({
+        namespace,
+        storage,
+        schema: {
+          port: {
```

Comment:
Are we missing `parser` part here? 

<img width="912" alt="Screenshot 2021-04-15 at 16 12 46" src="https://user-images.githubusercontent.com/639406/114883747-7b901f80-9e05-11eb-983e-8fc28047eca7.png">

Reply:
![image](https://user-images.githubusercontent.com/1761469/114884535-20126180-9e06-11eb-8eac-1ae51cd1cba0.png)

Strange… Are you on the correct TS version (I also try with `tsc` everything is compiling)